.github/workflows/automate.yml

```diff
@@ -0,0 +1,51 @@
+name: Java CI
+
+on: [push, pull_request]
+
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Set up JDK 21
+      uses: actions/setup-java@v2
+      with:
+        distribution: 'adopt'
+        java-version: '21'
+
+    - name: Run setup script
+      run: ./setup.sh
+
+    - name: Build and Test with Gradle
+      run: |
+        cd extractor
+        ./gradlew clean build test run --args="../.ci-temp/checkstyle" --info --stacktrace --console=plain \
+          -Dorg.gradle.logging.level=DEBUG \
+          -Dtest.single=MainsLauncherTest
+
+    - name: Check for changes
```

Comment:
Remove this step, it is not required

Reply:
how would it then find changes?

Reply:
No need to do this just commit all changes